pkg/kinflate/resource/resource.go

```diff
@@ -16,10 +16,57 @@ limitations under the License.
 
 package resource
 
-import "k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+import (
+	"encoding/json"
 
-// Examples: deployment, service, configmap, secret, etc.
-type Resource interface {
-	Name() string
-	Data() unstructured.Unstructured
+	"k8s.io/apimachinery/pkg/api/meta"
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/kubectl/pkg/kinflate/types"
+)
+
+// Resource represents a Kubernetes Resource Object for ex. Deployment, Server
+// ConfigMap etc.
+type Resource struct {
+	Data *unstructured.Unstructured
+
+	// This is primarily to unify the processing workflow for configmap with
+	// other resources. For configmap, we use hash of the payload as suffix for
+	// the name. This will be empty for other resources.
+	NameSuffix string
+}
+
+func (r *Resource) Name() string {
```

Comment:
I don't think we need the name if we have the GVKN

Reply:
agreed. will kill it.